src/main/java/com/github/messenger4j/send/templates/ListTemplate.java

```diff
@@ -0,0 +1,296 @@
+package com.github.messenger4j.send.templates;
+
+import com.github.messenger4j.internal.Assert;
+import com.github.messenger4j.send.buttons.Button;
+import com.google.gson.annotations.SerializedName;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * @author Jan Zarnikov
+ * @since 0.6.2
+ */
+public final class ListTemplate extends Template {
+
+    @SerializedName("top_element_style")
+    private final TopElementStyle topElementStyle;
+    private final List<Button> buttons;
+    private final List<Element> elements;
+
+    public static Builder newBuilder(TopElementStyle topElementStyle) {
+        return new Builder(topElementStyle);
+    }
+
+    private ListTemplate(Builder builder) {
+        super(TemplateType.LIST);
+        this.topElementStyle = builder.topElementStyle;
+        this.buttons = builder.buttons;
+        this.elements = builder.elements;
+    }
+
+    public enum TopElementStyle {
+        @SerializedName("large")
+        LARGE,
+
+        @SerializedName("compact")
+        COMPACT
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        if (!super.equals(o)) return false;
+        ListTemplate that = (ListTemplate) o;
+        return topElementStyle == that.topElementStyle &&
+                Objects.equals(buttons, that.buttons) &&
+                Objects.equals(elements, that.elements);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(super.hashCode(), topElementStyle, buttons, elements);
+    }
+
+    @Override
+    public String toString() {
+        return "ListTemplate{" +
+                "topElementStyle=" + topElementStyle +
+                ", buttons=" + buttons +
+                ", elements=" + elements +
+                '}';
+    }
+
+    public static final class Builder {
+
+        public static final int BUTTONS_LIMIT = 1;
+        private final TopElementStyle topElementStyle;
+
+        private List<Button> buttons;
+        private List<Element> elements;
+
+        public Builder(TopElementStyle topElementStyle) {
+            this.topElementStyle = topElementStyle;
```

Comment:
According the Messenger spec it is not required to provide a top element style (defaults to compact) but I think its a good practice to decide first what type of list template you want to create. Just a remark. No changes needed here.